include/envoy/request_info/request_info.h

```diff
@@ -158,6 +163,17 @@ class RequestInfo {
    * inferred from proxy proto, x-forwarded-for, etc.
    */
   virtual const Network::Address::InstanceConstSharedPtr& downstreamRemoteAddress() const PURE;
+
+  /**
+   * @return const Router::RouteEntry* Get the route entry selected for this request. Note: this
+   * will be nullptr if no route was selected.
+   */
+  virtual const Router::RouteEntry* routeEntry() const PURE;
+
+  /**
+   * Set the route entry once it's selected.
+   */
+  virtual void onRouteSelected(const Router::RouteEntry& route_entry) PURE;
```

Comment:
I wonder if this should be in a separate callback interface, since it's really an implementation detail of the request info, plus others might find that a useful callback to be able to hook?

Reply:
Hmmmm, I think that makes sense.  So this is something that the ActiveStream (or whoever owns the RequestInfo) would make available to the router/others?  Would we also want to package up `onUpstreamHostSelected()` in the same sort of interface?

Reply:
In this particular case, I actually think that connection manager can populate the route when it is loaded into the route cache variable, so, I would probably also remove this from the interface, and actually just set it directly from the connection manager, since it's useful to all filters.

Reply:
Good point. I didn't see that the route was actually being generated in the connection manager.  I'll move the population there. Thanks for pointing that out. @htuch, if we just move this method to `RequestInfoImpl` and call it from the `HttpConnectionManager`, will that be sufficiently flexible?  Or were you interested in adding this interface to some specific class of objects that the connection manager knows about?